cdap-common/bin/functions.sh

```diff
@@ -0,0 +1,1085 @@
+  local readonly __min=$(echo ${__node/v/} | cut -d. -f2)
+  local readonly __pat=$(echo ${__node/v/} | cut -d. -f3)
+  if [[ ${__maj} -lt $((__ver_maj + 1)) ]] && [[ ${__min} -lt $((__ver_min + 1)) ]] && [[ ${__pat} -lt $((__ver_pat + 1)) ]]; then
+    die "Node.js ${__node} is not supported. The minimum version supported is ${__ver}"
+    return 1
+  fi
+  return 0
+}
+
+#
+# cdap_get_conf <property> <conf-file> [default]
+# returns: property value if found, default if not found and default set, otherwise returns 1
+#
+cdap_get_conf() {
+  local readonly __pn=${1} __fn=${2} __default=${3} __result __property __sed_fu
+  # Check for xmllint
+  [[ $(which xmllint 2>/dev/null) ]] || {
+    case ${PLATFORM} in
+      RHEL) die "Cannot locate xmllint, is libxml2 installed?" ;;
+      UBUNTU) die "Cannot locate xmllint, is libxml2-utils installed?" ;;
+    esac
+    # If we get here, die
+    die "Cannot locate xmllint, are XML tools installed?"
+  }
+  # Get property from file, return last result, if multiple are returned
+  __property="cat //configuration/property[name='${__pn}']/value[text()]"
+  __sed_fu='/^\//d;s/^.*<value>//;s/<\/value>.*$//'
+  __result=$(echo "${__property}" | xmllint --shell "${__fn}" | sed "${__sed_fu}" | tail -n 1)
+  # Found result, echo it and return 0
+  [[ -n ${__result} ]] && echo ${__result} && return 0
+  # No result, echo default and return 0
+  [[ -n ${__default} ]] && echo ${__default} && return 0
+  return 1
+}
+
+#
+# cdap_kinit
+# Initializes Kerberos ticket using principal/keytab
+# 
+cdap_kinit() {
+  local readonly __principal=${CDAP_PRINCIPAL:-$(cdap_get_conf "cdap.master.kerberos.principal" "${CDAP_CONF}"/cdap-site.xml)}
+  local readonly __keytab=${CDAP_KEYTAB:-$(cdap_get_conf "cdap.master.kerberos.keytab" "${CDAP_CONF}"/cdap-site.xml)}
+  if [[ -z ${__principal} ]] || [[ -z ${__keytab} ]]; then
+    die "Both cdap.master.kerberos.principal and cdap.master.kerberos.keytab must be configured for Kerberos"
+  fi
+  if [[ ! -r ${__keytab} ]]; then
+    die "Cannot read keytab: ${__keytab}"
+  fi
+  if [[ $(which kinit 2>/dev/null) ]]; then
+    # Replace _HOST in principal w/ FQDN, like Hadoop does
+    kinit -kt "${__keytab}" "${__principal/_HOST/${HOSTNAME}}"
+    if [[ ! ${?} ]]; then
+      die "Failed executing 'kinit -kt \"${__keytab}\" \"${__principal/_HOST/${HOSTNAME}}\"'"
+    fi
+  else
+    die "Cannot locate kinit! Please, ensure the appropriate Kerberos utilities are installed"
+  fi
+  return 0
+}
+
+#
+# cdap_set_java
+# Attempts to find JAVA in few ways and sets JAVA variable
+#
+cdap_set_java () {
+  local readonly __java __java_version
+  # Check JAVA_HOME, first
+  if [[ -n ${JAVA_HOME} ]] && [[ -d ${JAVA_HOME} ]]; then
+    __java="${JAVA_HOME}"/bin/java
+    [[ -x ${__java} ]] || die "JAVA_HOME is set to an invalid location: ${JAVA_HOME}"
+  else
+    __java=${JAVA:-java}
+    if [[ $(which java 2>/dev/null) ]]; then
+      : # continue
```

Comment:
I tried to keep the same logic flow and the original source functions. I'll update this.